src/testbase.rs

```diff
@@ -13,13 +12,22 @@ use rand::rngs::StdRng;
 use rand::seq::SliceRandom;
 use rand::SeedableRng;
 
-use crate::aabb::{Bounded, AABB};
+use crate::aabb::Bounded;
 use crate::bounding_hierarchy::{BHShape, BoundingHierarchy};
-use crate::ray::Ray;
+
+// TODO These all need to be realtyped and bounded
```

Comment:
Is this TODO current?

Reply:
Yeah, I plan to expand the unit tests later to handle 2d and 4d cases

Reply:
Would you like to do that in this PR?

Reply:
@marstaik ping